File: IN1000/Obliger/Uke 10 og 11 peasent/verden.py
from rutenett import Rutenett

class Verden:
    def __init__(self, rader, kolonner):
        self._generasjonsnummer = 0
        
        self._rutenett = Rutenett(rader, kolonner)
        self._rutenett.fyll_med_tilfeldige_celler()
        self._rutenett.koble_celler()

    def tegn(self):
        self._rutenett.tegn_rutenett()
        print(f"Generasjon: {self._generasjonsnummer} - Antall levende celler: {self._rutenett.antall_levende()}")

    def oppdatering(self):
        for y in range(len(self._rutenett._rutenett)):
            for x in range(len(self._rutenett._rutenett[y])):
                self._rutenett._rutenett[y][x].tell_levende_naboer()
        for y in range(len(self._rutenett._rutenett)):
            for x in range(len(self._rutenett._rutenett[y])):
                self._rutenett._rutenett[y][x].oppdater_status()
        self._generasjonsnummer += 1

File: IN1010/Obliger/Oblig1/Verden.java
public class Verden {
    int antRader;
    int antKolonner;
    int genNr = 0;
    Rutenett rutenett;

    public Verden(int antRader, int antKolonner) {
        this.antRader = antRader;
        this.antKolonner = antKolonner;

        rutenett = new Rutenett(antRader, antKolonner);
        rutenett.fyllMedTilfeldigeCeller();
        rutenett.kobleAlleCeller();
    }

    public void tegn() {
        for (int i = 0; i < 10; i++) {
            System.out.println();
        }
        System.out.println("Generasjon nr " + genNr + ":");
        rutenett.tegnRutenett();
        System.out.println("\nDet er " + rutenett.antallLevende() + " levende celler.");
    }

    public void oppdatering() {
        for (int x = 0; x < rutenett.antRader; x++) {
            for (int y = 0; y < rutenett.antKolonner; y++) {
                rutenett.hentCelle(x, y).tellLevendeNaboer();
            }
        }
        for (int x = 0; x < rutenett.antRader; x++) {
            for (int y = 0; y < rutenett.antKolonner; y++) {
                rutenett.hentCelle(x, y).oppdaterStatus();
            }
        }
        genNr++;
    }


}
